articles/install-guide/index.md

```diff
@@ -1,53 +1,225 @@
-* [Microsoft Quantum Blog](https://cloudblogs.microsoft.com/quantum/?ext)
-* [Q# Blog](https://devblogs.microsoft.com/qsharp/)
+        HelloQ.simulate()
+        ```
 
+    - Run the program:
 
+        ```bash
+        python run_hello_q.py
+        ```
+
+    - Verify the output. Your program should output the following lines:
+
+        ```bash
+        Hello from quantum world!
+       0
+       ```
+
+## Develop with Jupyter notebooks
+
+1. Pre-requisites
+
+    - [Python](https://www.python.org/downloads/) 3.6 or later
+    - [Jupyter notebooks](https://jupyter.readthedocs.io/en/latest/install.html)
+
+1. Install the `iqsharp` package:
+
+    ```bash
+    conda install -c quantum-engineering iqsharp
+    ```
+
+1. Verify the installation by creating a `Hello World` application
+
+    - Run the following command to start the notebook server:
+
+        ```bash
+        jupyter notebook
+        ```
+
+    - Browse to the URL shown on the command line. For example: http://localhost:8888
+
+    - Create a jupyter notebook with a Q# kernel, and add the following code to the first notebook cell:
+
+        ```qsharp
+        operation HelloQ () : Unit {
+            Message($"Hello from quantum world!");
+        }
+        ```
+
+    - Run this cell of the notebook:
+
+        You should see `HelloQ` in the output of the cell. When running in jupyter notebooks, the Q# code is compiled, and the notebook outputs the name of the operation(s) that it finds.
+
+## Develop with C# on Windows, using Visual Studio
+
+1. Pre-requisites
+
+    - Visual Studio, version, with the .NET Core cross-platform development workload enabled
+
+1. Install the Q# Visual Studio extension
+
+    - Download the [Visual Studio extension](https://marketplace.visualstudio.com/items?itemName=quantum.DevKit)
+    - Add the extension to Visual Studio
+
+1. Verify the installation by creating a `Hello World` application
+
+    - Create a new Q# application
+
+        - Go to **File** -> **New** -> **Project**
+        - Type `Q#` in the search box
+        - Select **Q# Application**
+        - Select **Next**
+        - Choose a name and location for your application
+        - Select **Create**
+
+    - Inspect the project
+
+        You should see that two files have been created: `Driver.cs` which is the C# host application; and `Operation.qs`, which is a Q# program that defines a simple operation to print a message to the console.
+
+    - Run the application
+
+        - Select **Debug** -> **Start Without Debugging**
+        - You should see the text `Hello quantum world!` printed to a console window.
+
+## Develop with C#, using VS Code
+
+1. Pre-requisites
+
+   - [VS Code](https://code.visualstudio.com/download)
+   - [.NET Core SDK 2.1 or later](https://www.microsoft.com/net/download)
+
+1. Install the Quantum VS Code extension
+
+    - Download the [VS Code extension](https://marketplace.visualstudio.com/items?itemName=quantum.quantum-devkit-vscode)
```

Comment:
Personally, I don't tend to think of downloading and adding as separate steps, given that I don't save the extension to a VSIX before installing?

Reply:
Fixed